src/event/event.rs

```diff
@@ -78,6 +78,13 @@ impl Default for Event {
 }
 
 impl Event {
+    /// Returns an `Iterator` for `Attributes`
+    pub fn attributes_iter(&self) -> AttributesIter {
```

Comment:
I think you can change this signature to `impl Iterator<(&str, AttributeValue)>` or when you implement `ExactSizeIterator` `impl ExactSizeIterator<(&str, AttributeValue)>`. The reason is that we can keep private the implementation of the iterator, without exposing it as pub, and we can eventually change it in the future without breaking the public apis